openpaas-james/mailbox/encrypted/memory/src/main/java/com/linagora/openpaas/encrypted/InMemoryEncryptedMessageManager.scala

```diff
@@ -40,28 +42,37 @@ class InMemoryEncryptedMessageManager @Inject()(messageManager: MessageManager,
       .collectSeq()
       .block()
 
-    val out: ByteArrayOutputStream = new ByteArrayOutputStream
-    Encrypter.forKeys(keys.map(key => key.payload).asJava).encrypt(ByteSource.wrap(msgIn.readAllBytes()), out)
+    val out: FileBackedOutputStream = new FileBackedOutputStream(FILE_THRESHOLD)
+    try {
+      Encrypter.forKeys(keys.map(key => key.payload).asJava)
+        .encrypt(ByteSource.wrap(msgIn.readAllBytes()), out)
 
-    messageManager.appendMessage(new ByteArrayInputStream(out.toByteArray), internalDate, mailboxSession, isRecent, flags)
+      messageManager.appendMessage(out.asByteSource().openStream(), internalDate, mailboxSession, isRecent, flags)
+    } finally {
+      out.reset()
+    }
   }
 
   override def appendMessage(appendCommand: AppendCommand, session: MailboxSession): AppendResult = {
     val keys: Seq[PublicKey] = SFlux.fromPublisher(keystoreManager.listPublicKeys(session.getUser))
       .collectSeq()
       .block()
 
-    val out: ByteArrayOutputStream = new ByteArrayOutputStream
-    Encrypter.forKeys(keys.map(key => key.payload).asJava)
-      .encrypt(ByteSource.wrap(appendCommand.getMsgIn.getInputStream.readAllBytes()), out)
-
-    val encryptedCommand: AppendCommand = AppendCommand.builder()
-      .isRecent(appendCommand.isRecent)
-      .withFlags(appendCommand.getFlags)
-      .withInternalDate(appendCommand.getInternalDate)
-      .build(out.toByteArray)
-
-    messageManager.appendMessage(encryptedCommand, session)
+    val out: FileBackedOutputStream = new FileBackedOutputStream(FILE_THRESHOLD)
+    try {
+      Encrypter.forKeys(keys.map(key => key.payload).asJava)
+        .encrypt(new ByteSource { override def openStream(): InputStream = appendCommand.getMsgIn.getInputStream }, out)
```

Comment:
- Always override getSize when you implement bytesource so that it do not run the naive "fatch all" implementation...
 - Use `encrypt(MimeMessage)` here too...